woocommerce/woocommerce
# Add UTM parameters to links in subscription renewal notices

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds UTM parameters to subscription renewal links in notices shown on WC settings, in-app extensions and plugins page.

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

- Build WooCommerce on your local environment, e.g.: `cd plugins/woocommerce` then `npm run build:zip`.
- Update the WooCommerce plugin in your local store with WC build in previous step.
- Connected your store with woocommerce.test.
- Ensure Woo Update Manager is installed.



### Test Single Expired Subscription

- Ensure you have only one expired product key in woocommerce.test.
- Visit in-app my subscriptions tab (`/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions`)
- There should be a message like bellow and the renew button on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_in_apps_screen_renew
```

<img width="1655" alt="Screenshot 2024-07-17 at 19 56 29" src="https://github.com/user-attachments/assets/3542bc29-5e43-4ac9-b372-a45e6ac610ed">


- Visit  WC Settings
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_settings_screen_renew
```
<img width="858" alt="Screenshot 2024-07-17 at 20 15 14" src="https://github.com/user-attachments/assets/a8a85c7b-6b6a-4bad-8d64-282a2218b092">

- Visit plugin management page 
- Make sure the plugin version is below the current version by changing it manually on the plugin main file.
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_plugin_screen_renew
```
<img width="1047" alt="Screenshot 2024-07-17 at 20 21 09" src="https://github.com/user-attachments/assets/59e04f54-1214-427a-886b-142b4c0db553">



### Test Multiple Expired Subscriptions of different products

- Ensure you have 2 or more  expired product keys in woocommerce.test.
- Visit in-app my subscriptions tab (`/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions`)
- There should be a message like bellow and the renew button on the notice should have UTM parameters: 
```
utm_source=pu 
utm_campaign=pu_in_apps_screen_renew
```
<img width="766" alt="Screenshot 2024-07-17 at 20 30 46" src="https://github.com/user-attachments/assets/366e4b4a-a509-4da0-9fcc-b73a915dbbc2">

- Visit WC settings page
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_settings_screen_renew
```
<img width="757" alt="Screenshot 2024-07-17 at 20 32 26" src="https://github.com/user-attachments/assets/8fb29af7-7d3f-497a-b229-4c699d97ea3a">


### Test Multiple Expired Subscriptions of the same product

- Ensure you have 2 or more  expired product keys of same product (and no other expired keys) in woocommerce.test.
- Visit in-app my subscriptions tab (`/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions`)
- There should be a message like bellow and the renew button on the notice should have UTM parameters: 
```
utm_source=pu 
utm_campaign=pu_in_apps_screen_renew
```
<img width="781" alt="Screenshot 2024-07-17 at 20 45 55" src="https://github.com/user-attachments/assets/3def1e7e-19ff-4341-8cf4-179100de7e42">

- Visit WC settings page
- There should be a message like bellow and the renew link on the notice should have UTM parameters 
```
utm_source=pu 
utm_campaign=pu_settings_screen_renew
```
<img width="760" alt="Screenshot 2024-07-17 at 20 49 23" src="https://github.com/user-attachments/assets/219af60f-f8af-40c7-b307-b18ad742a9bb">





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add UTM parameters to subscription renewal notice links.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
